runtime/config-deployer-service/ballerina/APIClient.bal

```diff
@@ -809,33 +821,53 @@ public class APIClient {
+                model:HTTPRouteFilter headerModifierFilter = {'type: "RequestHeaderModifier"};
+                headerModifierFilter.requestHeaderModifier = self.extractHttpHeaderFilterData(requestPolicies, organization);
+                routeFilters.push(headerModifierFilter);
+            }
+            if responsePolicies is APKOperationPolicy[] && responsePolicies.length() > 0 {
+                model:HTTPRouteFilter headerModifierFilter = {'type: "ResponseHeaderModifier"};
+                headerModifierFilter.responseHeaderModifier = self.extractHttpHeaderFilterData(responsePolicies, organization);
+                routeFilters.push(headerModifierFilter);
+            }
         }
         return routeFilters;
     }
 
     isolated function extractHttpHeaderFilterData(APKOperationPolicy[] operationPolicy, commons:Organization organization) returns model:HTTPHeaderFilter {
+        model:HTTPHeader[] addPolicies = [];
         model:HTTPHeader[] setPolicies = [];
         string[] removePolicies = [];
         foreach APKOperationPolicy policy in operationPolicy {
-            string policyName = policy.policyName;
-
-            record {}? policyParameters = policy.parameters;
-            if (policyParameters is record {}) {
-                if (policyName == "addHeader") {
-                    model:HTTPHeader httpHeader = {
-                        name: <string>policyParameters.get("headerName"),
-                        value: <string>policyParameters.get("headerValue")
-                    };
-                    setPolicies.push(httpHeader);
+            if policy is HeaderModifierPolicy {
+                HeaderModifierPolicyParameters policyParameters = policy.parameters;
+                if policy.policyName == "AddHeaders" {
+                    ModifierHeader[] addHeaders = <ModifierHeader[]>policyParameters.headers;
+                    foreach ModifierHeader header in addHeaders {
+                        addPolicies.push(header);
+                    }
+                }
+                if policy.policyName == "SetHeaders" {
+                    ModifierHeader[] setHeaders = <ModifierHeader[]>policyParameters.headers;
+                    foreach ModifierHeader header in setHeaders {
+                        setPolicies.push(header);
+                    }
                 }
-                if (policyName == "removeHeader") {
-                    string httpHeader = <string>policyParameters.get("headerName");
-                    removePolicies.push(httpHeader);
+                if policy.policyName == "RemoveHeaders" {
+                    string[] removeHeaders = <string[]>policyParameters.headers;
+                    foreach string header in removeHeaders {
+                        removePolicies.push(header);
+                    }
                 }
             }
         }
         model:HTTPHeaderFilter headerModifier = {};
+        if (addPolicies != []) {
```

Comment:
```suggestion
        if addPolicies != [] {
```

Reply:
Fixed in https://github.com/wso2/apk/commit/3e75dbe7ea75f11d42daa95609e178756199f8e4